### DISCUSSION TITLE
Auto-Read Config File

### DISCUSSION THREAD

**Discussion by joshmedeski** [arl/gitmux#101] [Ideas] [answered]
I would love if gitmux could auto-detect it's config file with a standard file name and location.

It could be saved at `$HOME/.config/tmux/gitmux.conf` or `$HOME/.gitmux.conf`. This would simplify the tmux configuration as the config flag wouldn't need to be called (but could be overwritten if needed).

What do you think? I'm happy to contribute.

**Comment by arl** [comment, marked as answer]
That's a great idea. I've already thought about that also.

On the paper this is not a difficult feature, and it's surely useful. However there still are some things we should clearly define before starting:
 - the list of location that are searched for
 - in what order/precedence these locations are looked for

Once we have that, we're good.

Also, we'll need unit test that run for each platform, and to update the doc.

Not sure if you're familiar with Go, in case you're not feel free to ask questions.

Also, let's not reinvent the wheel here, I'm sure many projects use this scheme already (maybe even popular projects written in Go).

Thank you 🙏🏽
